frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -1002,7 +1001,6 @@ function ReaderBookmark:setBookmarkNote(item, from_highlight, is_new_note, new_t
         bookmark.type = self:getBookmarkType(bookmark)
         bookmark.text_orig = bm.text or bm.notes
         bookmark.mandatory = self:getBookmarkPageString(bm.page)
-        self.ui:handleEvent(Event:new("BookmarkEdited", bm))
```

Comment:
You removed one of my two BookmarkEdited. So, this one won't send any event, which kills my hunt for completeness of bookmark related events :) Unless it should really not be considered "bookmark edited" ?
May be you should just change the other BookmarkEdited to BookmarkUpdated and let this one be (and be not handled by ReaderThumbnail).
Or use a single BookmarkUpdate with an additional parameter visual_change=true/false.

Reply:
I think the event about editing the note must be sent when we press "Save" only, because when we press "Cancel" nothing changes.

Reply:
The one you removed on line 1005 is triggerered only when doing Cancel somewhere ? Looks like its trigerred when add/edit note in readerhighlight?
But obviously ok if it is like you say (can't look seriously at the code, trusting you).

Reply:
> Looks like its trigerred when add/edit note in readerhighlight?

When add/edit note dialog is shown.
Then we can press Cancel and no changes are saved.
So let's trigger the event within the Save callback only.

Reply:
So, you're saying that the following code path is taken only when the user hits Cancel somewhere - and it is then fine to remove line 1005 ?
https://github.com/koreader/koreader/blob/34c2dab54b479b7f11f898976476bb647497afc4/frontend/apps/reader/modules/readerbookmark.lua#L994-L1005

(It is not obvious that this is the case after a quick look, but trusting you if you think it really is :)

Reply:
No, this piece of code is a preparing part for showing the input dialog.
And the bookmark is "edited" only after pressing the Save button.

Reply:
OK, so it is just me that added `Event:new("BookmarkEdited")` here without thinking too much, and it was not needed, right ? :)

Reply:
Yes I think the event was added earlier than the bookmark is actually edited.